x/evm/wrappers/bank.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package wrappers
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/evmos/evmos/v20/x/evm/config"
+	"github.com/evmos/evmos/v20/x/evm/types"
+)
+
+var _ types.BankWrapper = BankWrapper{}
+
+// BankWrapper is a wrapper around the Cosmos SDK bank keeper
+// that is used to manage an evm denom with a custom decimal representation.
+type BankWrapper struct {
+	types.BankKeeper
+}
+
+// NewBankWrapper creates a new BankWrapper instance.
+func NewBankWrapper(
+	bk types.BankKeeper,
+) *BankWrapper {
+	return &BankWrapper{
+		bk,
+	}
+}
+
+// ------------------------------------------------------------------------------------------
+// Bank wrapper own methods
+// ------------------------------------------------------------------------------------------
+
+// MintAmountToAccount converts the given amount into the evm coin scaling
+// the amount to the original decimals, then mints that amount to the provided account.
+func (w BankWrapper) MintAmountToAccount(ctx context.Context, recipientAddr sdk.AccAddress, amt math.Int) error {
+	coin := sdk.NewCoin(config.GetEVMCoinDenom(), amt)
+
+	convertedCoin, err := convertEvmCoinFrom18Decimals(coin)
+	if err != nil {
+		return errors.Wrap(err, "failed to mint coin to account in bank wrapper")
+	}
+
+	coinsToMint := sdk.NewCoins(convertedCoin)
+	if err := w.MintCoins(ctx, types.ModuleName, coinsToMint); err != nil {
+		return errors.Wrap(err, "failed to mint coins to account in bank wrapper")
+	}
+
+	return w.SendCoinsFromModuleToAccount(ctx, types.ModuleName, recipientAddr, coinsToMint)
+}
+
+// BurnAmountFromAccount convert the given amount into the evm coin scaling
+// the amount to the original decimals, then burns that quantity from the provided account.
+func (w BankWrapper) BurnAmountFromAccount(ctx context.Context, account sdk.AccAddress, amt math.Int) error {
+	coin := sdk.NewCoin(config.GetEVMCoinDenom(), amt)
+
+	convertedCoin, err := convertEvmCoinFrom18Decimals(coin)
+	if err != nil {
+		return errors.Wrap(err, "failed to burn coins from account in bank wrapper")
+	}
+
+	coinsToBurn := sdk.NewCoins(convertedCoin)
+	if err := w.SendCoinsFromAccountToModule(ctx, account, types.ModuleName, coinsToBurn); err != nil {
```

Comment:
the amount here is already scaled, so we need to use the original bankKeeper method
```suggestion
	if err := w.BankKeeper.SendCoinsFromAccountToModule(ctx, account, types.ModuleName, coinsToBurn); err != nil {
```

Reply:
Damn, pretty sure to have changed it..thanks for spotting it!